### PR TITLE
haruna: 0.7.3 -> 0.8.0

### LINKED ISSUE #169970: haruna: request update 0.7.3 -> 0.8.0
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: haruna

current version: 0.7.3
desired version: 0.8.0

###### Notify maintainers

maintainers: @jojosch 

###### Note for maintainers

I tried to update it myself, but when trying to open the result I get either:
```
fish: Job 1, '/nix/store/qjwn854dkqnbbx6pc6aw…' terminated by signal SIGSEGV (Address boundary error)
```
or:
```
unknown() : QQmlApplicationEngine failed to load component

unknown() : qrc:/qml/main.qml:78:5: SettingsEditor is not a type
```

Some dependencies could be moved to `nativeBuildInputs`, since they are build time only.

The project supports `yt-dlp` other than `youtube-dl`. Since `mpv` uses the former, maybe switching could be considered.

The homepage is outdated. It's now "https://invent.kde.org/multimedia/haruna".

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by SuperSamus**
###### Description of changes
Supersedes #171328.
Closes #169970 (which contains info on what's done here).

For some reason, when I try to open the resulting binary, I get a segfault (see the issue above), whether it's a simple version bump or this PR. It's the reason I opened it as draft.
However there seems to be no problem to the person in the PR I linked, so I don't know if this is a problem in my end...
(TL;DR, test whether it opens on your end, if it does then it won't be draft anymore.)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSamus** [comment]
Ok, I tried to investigate, like I said on the issue above, sometimes the error that comes when opening Haruna is:
```
unknown() : QQmlApplicationEngine failed to load component

unknown() : qrc:/qml/main.qml:78:5: SettingsEditor is not a type
```
But `SettingsEditor` doesn't exist anymore: https://invent.kde.org/multimedia/haruna/-/commit/c7f0fa831e42e6d4b4692abde6e1634dc7f0a6f5.
Does anyone know what is going on?
Tried to do the workaround on https://github.com/NixOS/nixpkgs/issues/177720, didn't solve anything.

**Comment by hqurve** [comment]
For me, the workaround worked. Steps I did:

1. Run haruna from 6616de389ed55fba6eeba60377fc04732d5a207c (my system's current nixpkgs)
2. Fetch this pull request and build the binary
3. `./result/bin/haruna` results in a segmentation fault
4. Delete ~/.cache/kde (since I could not have found anything called haruna in my cache directory, I looked at the times modified and I got lucky with my guess)
5. `./result/bin/haruna` launches as expected

Also, if I try to load haruna from the older commit, it also fails to run, albeit with a different error,
```
unknown() : QQmlApplicationEngine failed to load component

unknown() : qrc:/qml/main.qml:20:1: "Menus/Global": no such directory
```
However, clearing the same ~/.cache/kde fixes the issue.

Also, also, I repeated the steps on the same machine again after going back to the later commit and it works.

**Comment by SuperSamus** [comment]
Clearing ~/.cache/kde worked for me too.
I'll open this then.
Also fixes https://github.com/NixOS/nixpkgs/issues/170026 (I don't reproduce it anymore).

File: pkgs/applications/video/haruna/default.nix [hunk 1]
```diff
@@ -6,69 +6,68 @@
...
+    wrapQtAppsHook
+
     ffmpeg-full
-    kcodecs
     kconfig
     kcoreaddons
     kfilemetadata
     ki18n
     kiconthemes
     kio
```
Comment (SuperSandro2000) [inline pkgs/applications/video/haruna/default.nix:58]: Half of these are at least buildInputs

Comment (SuperSamus) [reply pkgs/applications/video/haruna/default.nix:51]: According to the GitLab page, they are only needed at build time.
https://invent.kde.org/multimedia/haruna/-/blob/master/README.md#dependencies

File: pkgs/applications/video/haruna/default.nix [hunk 2]
```diff
@@ -6,69 +6,68 @@
...
-  version = "0.7.3";
+  version = "0.8.0";
 
   src = fetchFromGitLab {
     owner = "multimedia";
     repo = "haruna";
     rev = "v${version}";
-    sha256 = "sha256-pFrmTaRvsqxJw34VULzfjx2k56kJgkB96nJtai2D1wY=";
+    sha256 = "sha256-Lom9iQUKH3lQHrVK4dJzo+FG79xSCg0b4gY/KAevL6I=";
     domain = "invent.kde.org";
```
Comment (SuperSandro2000) [inline pkgs/applications/video/haruna/default.nix:36]:
```suggestion
```

File: pkgs/applications/video/haruna/default.nix [hunk 3]
```diff
@@ -6,69 +6,68 @@
...
-, youtube-dl
+, yt-dlp
 }:
 
 mkDerivation rec {
   pname = "haruna";
-  version = "0.7.3";
+  version = "0.8.0";
 
   src = fetchFromGitLab {
```
Comment (SuperSandro2000) [inline pkgs/applications/video/haruna/default.nix:31]:
```suggestion
  src = fetchFromGitLab {
    domain = "invent.kde.org";
```

File: pkgs/applications/video/haruna/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/video/haruna/default.nix:51]: Yes, before you had them in nativeBuildInputs which is not what we want.

**Comment by SuperSamus** [comment]
I also don't reproduce #170026 anymore after this.

**Pull request merged**